assembly/src/v1/tests.rs

```diff
@@ -107,6 +107,58 @@ fn script_with_nested_procedure() {
     assert_eq!(expected, format!("{}", script));
 }
 
+// COMMENTS
+// ================================================================================================
+
+#[test]
+fn comment_simple() {
+    let assembler = super::Assembler::new();
+    let source = "begin # simple comment # push.1 push.2 add end";
```

Comment:
In the current state of the `next` branch, `push.1` gets actually expanded into `pad incr` - so, this test might not pass. Could you rebase from `next` and see if all still works correctly?

Reply:
I rebased from next and both tests fail with this output:

---- v1::tests::comment_simple stdout ----
thread 'v1::tests::comment_simple' panicked at 'assertion failed: `(left == right)`
  left: `"begin span push(1) push(2) add end end"`,
 right: `"begin span pad incr push(2) add end end"`', assembly/src/v1/tests.rs:119:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- v1::tests::comment_in_nested_control_blocks stdout ----
thread 'v1::tests::comment_in_nested_control_blocks' panicked at 'assertion failed: `(left == right)`
  left: `"begin join join span push(1) push(2) end if.true join span add end while.true span push(7) push(11) add end end end else join span mul push(8) push(8) end if.true span mul end else span noop end end end end end span push(3) add end end end"`,
 right: `"begin join join span pad incr push(2) end if.true join span add end while.true span push(7) push(11) add end end end else join span mul push(8) push(8) end if.true span mul end else span noop end end end end end span push(3) add end end end"`', assembly/src/v1/tests.rs:159:5

Reply:
Yeah, the expected result for both just needs to be changed from push(1) to pad incr as mentioned above. I'll update it soon, was just finishing up a few other things :)

Reply:
Could maybe we use the TokenStream itself for the test? I'm still a Rust beginner and don't know how to do that exactly but I tried something like that: 

```
let source = "begin # simple comment # push.1 push.2 add end";
let tokenStream = match super::TokenStream::new(source) {
    Ok(tokenStream) => tokenStream,
    Err(_) => panic!(),
};
```
I don't know how to get all the tokens from the tokenStream, but if I print it I get this array:

`["begin", "push.1", "push.2", "add", "end"]`

Then I guess we could just assert the result is this array. What do you think? ;)

Reply:
Sure, that would avoid needing changes to the tests if the compilation output for tokens in the test changes. In this case, I followed the pattern that Bobbin already had in place for testing, and I do like doing the comparison after compiling fully because that way if the implementation moves in the future then the test will still be a good test.

Reply:
Yes - we could refactor the tests to make some of them more self-contained within the `TokenStream` module - but I don't think we need to do that in this PR. Could be a good task for another entry-level PR though.